src/NHibernate/Async/Engine/Collections.cs

```diff
@@ -204,6 +204,7 @@ private static Task PrepareCollectionForUpdateAsync(IPersistentCollection collec
 				{
 					// it is or was referenced _somewhere_
 					bool ownerChanged = loadedPersister != currentPersister ||
+					!(entry.LoadedKey is DelayedPostInsertIdentifier) &&
```

Comment:
When an entity owning some collection is persisted then flushed, the collection entry stores the entity id as being its loaded key. In case of delayed identifier insert, it stores the `DelayedPostInsertIdentifier` instance for that entity.
If flushing again, the collection entry current key will then be the entity actual identifier, which will of course not match the `DelayedPostInsertIdentifier` instance previously associated with the entity.

So when the loaded key is a `DelayedPostInsertIdentifier`, we can not know whether the owner is still the same or not. This change just assume the collection owner has not actually changed in such case, as done in HHH-12826.

So here, if the collection owner was actually changed with a collection supporting it (not one with delete-orphan), we have a regression: previously it would have correctly transferred the children to the new owner.

A better fix would be to have the actual identifier stored as loaded key in the collection entry. Here is the detailed sequence of events storing the loaded key in above case:
- On Persist, the identifier of the owner is set as the collection entry current key. There, it could be a `DelayedPostInsertIdentifier` due to this PR.
- On Flush, the entry loaded key is set with the entry current key (in `CollectionEntry.AfterAction`). Maybe at that point could the actual identifier be retrieved instead.